pkg/model/iam.go

```diff
@@ -54,6 +54,13 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	// Collect the roles in use
 	var roles []kops.InstanceGroupRole
 	for _, ig := range b.InstanceGroups {
+		if ig.Spec.IAM != nil && ig.Spec.IAM.Profile != nil {
```

Comment:
So using this pattern where we do not create the tasks bypasses our validation for shared objects.  We still need to create the tasks but mark the tasks as shared. Take a look at how I am doing that in https://github.com/kubernetes/kops/pull/3625.  Please ping me if you have questions.  We will need to create shared roles, profiles, etc.  Also, delete kind bugs me since we will not have shared tags, please let me know how delete is working.

Looks great, we are really close.

Reply:
The `buildShareIAMInstanceProfile` function is already creating an `IAMInstanceProfile` task marked as shared. You're saying I need to also create a shared `IAMInstanceProfileRole` task? I can get that added, as well as test deletion behavior.

Also, do you think its worth it to add a validation warning/error when both `additionalPolicies` and `.iam.profile` are set?

Reply:
Yes we need the role and policy tasks. @justinsb is pretty passionate about that. 

In regards to the delete. Kops will try to delete the profile of the name is the same as what kops typically creates. I think that is a limitation we will need to live with.

Reply:
One more scenario I’m unsure about: if an existing cluster is updated to use this feature, will the original roles ever get deleted? The first ‘update cluster’ can’t delete them because they’ll be in use until a ‘rolling update’ has been ran at which point Kops won’t even know about the original roles. 

Shouldn’t deletion reference the cluster and IG specs, determine the roles are shared, and not to look for roles based off the original naming scheme?